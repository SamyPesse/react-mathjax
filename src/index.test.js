import React from 'react'
import ReactDOM from 'react-dom'
import * as MathJax from './'
import renderer from 'react-test-renderer'

it('Renders correctly', ()=>{
  const div = document.createElement('div')
  const tex = `f(x) = \\int_{-\\infty}^\\infty
      \\hat f(\\xi)\\,e^{2 \\pi i \\xi x}
      \\,d\\xi`

  const example = (
    <MathJax.Context>
      <div>
        This is an inline math formula: <MathJax.Node inline>{'a = b'}</MathJax.Node>
        And a block one:

        <MathJax.Node>{tex}</MathJax.Node>
      </div>
    </MathJax.Context>
  )
  ReactDOM.render(example, div)

  const tree = renderer.create(example).toJSON()
  expect(tree).toMatchSnapshot()
})

