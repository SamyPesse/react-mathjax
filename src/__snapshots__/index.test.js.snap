// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  This is an inline math formula: 
  <span />
  And a block one:
  <span />
</div>
`;
